### PR TITLE
Weimer2005 empirical

### PR DESCRIPTION
**Pull request by ZhuxiaoLi66**
The IPE branch includes the new version of empirical model Weimer2005 into ipelib/efield_empirical of IPE. The error codes has been returned for file wrongly reading and some 'stop' comments has been removed from the weimer code.

### CODE REVIEW

**Comment by twfang** [comment]
@akubaryk please quickly read through the changes, to see if there are any concerns. I have been carried out runs with the branch to see if there are any problems. Thanks!

File: src/ipelib/empirical_efield/efield_ipe.f [hunk 1]
```diff
@@ -250,66 +262,42 @@ subroutine get_efield(i_rank)
...
 !-----------------------------------------------------------------------
       call DerivPotential
-!     print*,'ed2_efield',ed2(149,65),potent(149,66),potent(149,64)
 
-      end subroutine get_efield
+!      if(i_rank.eq.0) then
+!           write(2000+i_rank, *) potent
+!           write(2000+i_rank, *) ed1 
+!           write(2000+i_rank, *) ed2
+!      end if
```
Comment (akubaryk) [inline src/ipelib/empirical_efield/efield_ipe.f]: overall the code changes read fine, would just clean up some of the new commented debug blocks in efield_ipe.f -- if there are print or write statements that are not needed generally, they should not be included in the pull request.

File: src/ipelib/empirical_efield/efield_ipe.f [hunk 2]
```diff
@@ -250,66 +262,42 @@ subroutine get_efield(i_rank)
...
 ! ajust S_a
 !-----------------------------------------------------------------------
-      call adj_S_a(i_rank)
+      call adj_S_a
 !-----------------------------------------------------------------------
 ! calculate global electric potential    
 !-----------------------------------------------------------------------
-      call GlobalElPotential(i_rank)
+      call GlobalElPotential
+!     print*,'pot_efield',potent(149,66),potent(149,64)
```
Comment (akubaryk) [inline src/ipelib/empirical_efield/efield_ipe.f:285]: unnecessary comment

File: src/ipelib/empirical_efield/efield_ipe.f [hunk 3]
```diff
@@ -1019,26 +976,50 @@ subroutine prep_weimer
...
+!!      if( by == 0. .and. bz == 0.) then
+!!         angle = 0.
+!!      else
+!!         angle = atan2( by,bz )
+!!      end if
+!!      angle = angle*rtd
+!!      call adjust_ipe( angle )
+!!      bt = sqrt( by*by + bz*bz )
+
+!!        print *, angle,bt,tilt,v_sw,swden
```
Comment (akubaryk) [inline src/ipelib/empirical_efield/efield_ipe.f:1008]: drop this large block

File: src/ipelib/empirical_efield/efield_ipe.f [hunk 4]
```diff
@@ -1537,6 +1464,7 @@ subroutine interp_poten( pot_highlats, pot_highlat, pot_midlat,
         if (j2 < 0) j2 = 0              ! Tomoko's fix - j2 >= 0
 	wrk1 = pot_midlat(ilon,j1)
 	wrk2 = pot_highlats(ilon,j2)
+!        write(iulog,*) 'pot_all ',ilon,hb1,hb2,nmlath -ibnd,tw
```
Comment (akubaryk) [inline src/ipelib/empirical_efield/efield_ipe.f:1461]: unnecessary comment

File: src/ipelib/empirical_efield/efield_ipe.f [hunk 5]
```diff
@@ -2842,8 +2120,9 @@ SUBROUTINE GECMP (AMLA,RMLT,ET,EP)
 	AMLA1 = 180. - AMLA1
 	XMLT1 = XMLT1 + 12.
       ENDIF
-      P1 = EPOTVAL (AMLA1    ,XMLT1)
-      P2 = EPOTVAL (AMLA-STPD,XMLT )
+      call EpotVal_new(AMLA1    , XMLT1, P1 )
+      call EpotVal_new(AMLA-STPD, XMLT,  P2 )
+!      print *, "gecmp_ipe ", P1, P2, AMLA1, XMLT1, AMLA-STPD, XMLT
```
Comment (akubaryk) [inline src/ipelib/empirical_efield/efield_ipe.f:2119]: unnecessary comment

File: src/ipelib/empirical_efield/efield_ipe.f (hunk 1 above)
Comment (ZhuxiaoLi66) [reply src/ipelib/empirical_efield/efield_ipe.f]: already deleted.

File: src/ipelib/empirical_efield/efield_ipe.f (hunk 2 above)
Comment (ZhuxiaoLi66) [reply src/ipelib/empirical_efield/efield_ipe.f:285]: for future coding and debug

File: src/ipelib/empirical_efield/efield_ipe.f (hunk 3 above)
Comment (ZhuxiaoLi66) [reply src/ipelib/empirical_efield/efield_ipe.f:1008]: for reminder and future coding

File: src/ipelib/empirical_efield/efield_ipe.f (hunk 4 above)
Comment (ZhuxiaoLi66) [reply src/ipelib/empirical_efield/efield_ipe.f:1461]: in old code and found it was helpful for my coding, so that for others

File: src/ipelib/empirical_efield/efield_ipe.f (hunk 5 above)
Comment (ZhuxiaoLi66) [reply src/ipelib/empirical_efield/efield_ipe.f:2119]: same thing as above

**Comment by twfang** [comment]
I have tested the branch with all three storms. The code runs fine. I didn't see any problems in the results. I think we can get it merged in.
